src/transformStateWithClones.js

```diff
@@ -65,7 +65,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const transformatedState = [];
+  const changedProperties = { ...state };
+
+for (const transformation of transforms) {
+  const { properties, operation } = transformation;
+
+  switch (operation) {
+    case 'addProperties':
+      Object.assign(changedProperties, properties);
+      transformatedState.push({ ...changedProperties });
+      break;
+
+    case 'removeProperties':
+      for (const key of properties) {
+        delete changedProperties[key];
+      }
+      transformatedState.push({ ...changedProperties });
+      break;
+
+    case 'clear':
+      for (const prop in changedProperties) {
+        delete changedProperties[prop];
+      }
+      transformatedState.push({ ...changedProperties });
```

Comment:
now you write push 3 times in each if, let's push changes only once on each loop iteration